11.4 RNs (#10253)
mendix/docs: content/en/docs/releasenotes/studio-pro/11/11.4.md

* We improved the rendering performance of repeating widgets in Design mode, by only rendering a single repeat for widgets within a snippet call, parent layout, or nested within another repeating widget.
* We improved the consumed REST service editor in Studio Pro. It is now less strict in the names of parameters, allowing all characters except `:/?#@!$&'()*+,;=\"` and spaces.
* We improved the OpenAPI document for published OData services:
    * It no longer specifies the `style` property of parameters when it is the default style. This makes the document shorter without changing its semantics.
    * It now indicates whether responses are optional or not.
* In the Consumed REST Service document in Studio Pro you can now import OpenAPI/Swagger contracts that contain errors. This is especially useful when the errors are in parts of the contract that the app does not use.
* The Mendix runtime will no longer start if an invalid time zone is set. This was a critical log message before.
* We added the `AbsoluteSessionTimeout` runtime setting. When this setting is configured, sessions will be expired regardless of whether they are active or not.

> * We now hide unsupported widgets from the toolbox when React client is enabled.

Reviewer: ```suggestion
* We now hide unsupported widgets from the toolbox when the React client is enabled.
```
"React client" seems to have a definite article in other parts of this document.

* We have optimized the page exports for the React and Native clients. This can decrease the exported page size by up to 35%, making the resulting bundle a lot smaller.
* You can now drag a consumed REST service to a microflow. This creates a *Send REST Request* activity.

### Fixes

> * We fixed an issue where a dropdown reopened automatically after closing. (Tickets 246215, 257081, 257895, 258443, and 261466)

Reviewer: Per the style guide, "drop-down" is the correct spelling. Additionally, it **must not** be used as a noun.

Reviewer 2: Thanks for the reminder.
```suggestion
* We fixed an issue where a drop-down list in a Reference Selector reopened automatically after closing. (Tickets 246215, 257081, 257895, 258443, and 261466)
```

* We fixed an issue in the React client where autofocus was incorrectly triggered by dynamically loaded widgets, causing combo boxes that use a widget with a dynamic source in their custom content to unexpectedly lose focus and close. (Ticket 254768)
* We fixed an issue in the Studio Pro XPath constraint parser that incorrectly allowed associations and associated objects to be compared to Boolean values or empty, while that was not supported on the runtime. The following constructs will now lead to consistency errors:
    * `[MyModule.MyAssociation = false()]`
    * `[MyModule.MyAssociation = empty]`
    * `[MyModule.MyAssociation/MyModule.MyEntity = false()]`
    * `[MyModule.MyAssociation/MyModule.MyEntity = empty]`

    (Tickets 254830 and 258029)
* We fixed an issue where Import mappings using 'Find an object (by key)' or 'Call a microflow' did not create an object for valid input. (Tickets 255130 and 260462)
* We fixed an issue where trying to delete a huge number (millions) of anonymous sessions during startup would cause out of memory errors. We now delete sessions and anonymous users in batches to avoid this problem. (Ticket 257047)

> * We fixed an issue where Node.js would run out of memory during the bundling process of very large, complex applications using the React client (Ticket 257472)

Reviewer: ```suggestion
* We fixed an issue where Node.js would run out of memory during the bundling process of very large, complex applications using the React client. (Ticket 257472)
```

* We fixed an issue where the React client would not get bundled when deploying for Eclipse. (Ticket 258470)
* In the logic editors, we fixed the zoom in/zoom out behavior of error handling icons. (Ticket 258424)
* We improved the performance of the tree control that allows you to select elements for Message Definitions and Import/Export Mappings. (Ticket 258606).
* When using Oracle, we no longer use `dbms.lob_substr` in the generated SQL for enumeration attributes with values shorter than 2000 characters. (Ticket 259283)
* We fixed an issue where a Java action could still cause compilation errors, even if excluded from the project. (Ticket 259414)
* We fixed an issue with offline PWAs when using the React client where changes to the application would not be shown to the user without clearing the cache. (Ticket 260673)
* We fixed a "no such column" error when synchronizing a file or image derived entity which itself is derived by another entity with an offline synchronization mode. (Ticket 261700)
* We fixed an issue in offline apps which caused "no such table" errors when the owning entities of associations used in XPath constraints were not offline entities. (Ticket 258840)

> * We fixed an issue in offline apps which caused "no such table" errors when an entity had an online synchronization mode specified in the synchronization nanoflow action.

Reviewer: I would combine the two items referring to the "no such table" error as follows:

```suggestion
* We fixed an issue in offline apps which caused "no such table" errors in the following scenarios:
 
   * The owning entities of associations used in XPath constraints were not offline entities. (Ticket 258840)
   * An entity had an online synchronization mode specified in the synchronization nanoflow action.
```

Reviewer 2: Nice suggestion - I've had to redo it as the change should be for two lines and not just one.
But I have just copy/pasted your suggestion.

> * We fixed an issue in offline apps which caused "no such table" errors when an entity had an online synchronization mode specified in the synchronization nanoflow action.

Reviewer 2: ```suggestion
* We fixed an issue in offline apps which caused "no such table" errors in the following scenarios:
 
   * The owning entities of associations used in XPath constraints were not offline entities. (Ticket 258840)
* An entity had an online synchronization mode specified in the synchronization nanoflow action.
```

* We fixed an issue where duplicate responsive navigation profiles could not be removed.
* We fixed an issue in the Studio Pro XPath constraint parser with the treatment of true and false . These are now treated the same as their function equivalents (true() and false() respectively).
* We fixed an issue where events created by the sidebar toggle button weren't getting cleaned up, leading to memory issues.
* We fixed an issue in the modernized app explorer where the search query would not reset when opening or creating a new app.
* We fixed checking version functionality that allowed you to perform version control operations on revisions that were created on older Studio Pro versions, or which had invalid metadata.
    Affected operations are: rebase, merge, merge feature branch, cherry-pick and revert.
* We fixed an issue in workflows where a multi-user task would not fail if it has no targeted users.
* We fixed an issue in offline apps that could  result in data validations of microflow calls not being properly handled.
* We fixed a memory leak that occurred when editing an action of a Microflow.

> * We fixed a memory leak when re-opening a project, or switching branches in Studio Pro.

Reviewer: ```suggestion
* We fixed a memory leak that occurred when re-opening a project, or switching branches in Studio Pro.
```

* We fixed an issue in Studio Pro where restarting Windows with an open app and then opening the app again would result in a warning dialog about the app already being open.
* We fixed an issue with opening a Studio Pro 10 app that uses the send REST request microflow activity in Studio Pro 11. For operations with query parameters this would result in a consistency error about the parameters having changed.
* We fixed an issue when using Send REST Activity where refreshing parameters in the Parameters grid did not update the Base URL.
* We fixed an issue in the Call External Action microflow activity. To call an action without arguments, it now sends `{}` in the request body, rather than having an empty request body.
* We fixed an issue where task queue tasks were not able to mark themselves as completed after execution.
* We fixed an issue where the object version attribute incorrectly incremented when an unchanged object is committed.
* We fixed an issue with using a dash `-` in the *Organization Name* of a pluggable widget.

> * We fixed an issue in the Domain Model editor where navigating to a specific OQL view entity with a consistency error from the Error pane would result in an error if the "Show View Entity" option was disabled.

Reviewer: ```suggestion
* We fixed an issue in the Domain Model editor where navigating to a specific OQL view entity with a consistency error from the **Error** pane would result in an error if the **Show View Entity** option was disabled.
```
